src/sources/host_metrics.rs

```diff
@@ -692,6 +694,22 @@ fn add_collector(collector: &str, mut metrics: Vec<Metric>) -> Vec<Metric> {
     metrics
 }
 
+#[cfg(target_os = "linux")]
+pub fn init_roots() {
+    let virt = futures::executor::block_on(heim::virt::detect());
+
+    if let Some(virtualized) = virt {
+        if virtualized.is_container() {
+            if let Some(procfs_root) = std::env::var_os("PROCFS_ROOT") {
+                heim_runtime::linux::set_procfs_root(PathBuf::from(&procfs_root));
+            }
+            if let Some(sysfs_root) = std::env::var_os("SYSFS_ROOT") {
+                heim_runtime::linux::set_sysfs_root(PathBuf::from(&sysfs_root));
+            }
+        }
+    }
```

Comment:
Is this feature useful for non-virtual environments, like for doing testing? That is, would there be any problems with removing the `virtualized` guard here?

Reply:
I can't imagine that it would be super useful? But, I could be wrong!

Reply:
Let’s not apply additional constraints - it’s already an opt-in - it makes sense to keep it simple to make it more predictable.

Reply:
Btw, containers are sort of a well known use case for this, but I’ve seen special dev environments with custom linux-based OS where the procfs and sysfs were mounted in non-standard locations. It’s hardly a case for Vector production use, but there might be users with such a setup out there.